optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/tenant/RosterParametrization.java

```diff
@@ -34,6 +34,14 @@
     @NotNull
     private Integer rotationEmployeeMatchWeight = 500;
     @NotNull
+    private Integer lowRiskEmployeeInCovidWardMatchWeight = 1; // Soft
+    @NotNull
+    private Integer moderateRiskEmployeeInCovidWardMatchWeight = 5; // Soft
+    @NotNull
+    private Integer highRiskEmployeeInCovidWardMatchWeight = 10; // Soft
+    @NotNull
+    private Integer extremeRiskEmployeeInCovidWardMatchWeight = 1; // Hard
+    @NotNull
```

Comment:
I have noticed that there is a `...View` version of this class, which is more or less identical to this one.
What is the purpose of that one, and do I need to duplicate the code there?

Reply:
@cuijulian Why do we have a separate view class for `RosterParametrization`? I don't see anything that need to be serialized differently for the client.

Reply:
(we don't call getRosterParametrization/updateRosterParametrization in the client yet, so you do not need to update it unless we want to add a config screen.

Reply:
Pretty much all domain entities have a `...View` version. This was for Spring security reasons iirc, it wasn't good practice to pass entities directly to RestControllers, so the `...View` objects are passed instead, which then get converted to the non-view entities.

Reply:
Why not create a new Entity and update the fields manually?
(i.e:
```
       oldRosterParametrization.setDesiredTimeSlotWeight(updatedRosterParametrization.getDesiredTimeSlotWeight());
        oldRosterParametrization.setRotationEmployeeMatchWeight(
                updatedRosterParametrization.getRotationEmployeeMatchWeight());
        oldRosterParametrization.setUndesiredTimeSlotWeight(
                updatedRosterParametrization.getUndesiredTimeSlotWeight());
        oldRosterParametrization.setWeekStartDay(updatedRosterParametrization.getWeekStartDay());
```

Reply:
(we are not persisting an entity directly, but updating the fields of an entity)

Reply:
What would be passed into the RestController in that case? Wouldn't you still need to pass in the `updatedRosterParametrization` (which is still an `@Entity`) in the PUT request?

Reply:
Code will look like this:
```
createEntity(Entity newEntity) {
    Entity entity = new Entity();
    // verify legal
    entity.setX(newEntity.getX());
    // ...
    persist(entity);
}

updateEntity(Entity updatedEntity) {
    Entity oldEntity = getOldEntity();
    // verify legal
    oldEntity.setX(updatedEntity.getX());
    // ...
    persist(oldEntity)
}
```